tests/e2e/autoscaling/autoscaler.go

```diff
@@ -295,25 +305,29 @@ var _ = Describe("Cluster size autoscaler", Label(utils.TestSuiteLabelFeatureAut
 		err = utils.WaitPodsToBeReady(cs, ns.Name)
 		Expect(err).NotTo(HaveOccurred())
 
-		By("Scaling up, 25 nodes at a time until reaches 50 nodes")
+		By("Scaling up, 25 nodes at a time until reaches 50 nodes (38 nodes for aks cluster)")
 		cpu := nodes[0].Status.Capacity[v1.ResourceCPU]
 		scaleUpPodSize := int64(float64(cpu.MilliValue()) / 1.8)
 		scaleUpDeployment := createDeploymentManifest(basename+"-deployment1", 0, map[string]string{"app": basename + "-deployment1"}, scaleUpPodSize, false)
 		_, err = cs.AppsV1().Deployments(ns.Name).Create(context.TODO(), scaleUpDeployment, metav1.CreateOptions{})
 		Expect(err).NotTo(HaveOccurred())
 
 		targetNodeCount := initNodeCount
+		expectedNode := 50
+		if os.Getenv(utils.AKSTestCCM) != "" {
+			expectedNode = 38
+		}
 		for {
-			*scaleUpDeployment.Spec.Replicas += 25
-			targetNodeCount += 25
+			*scaleUpDeployment.Spec.Replicas += int32(expectedNode / 2)
+			targetNodeCount += expectedNode / 2
 			_, err = cs.AppsV1().Deployments(ns.Name).Update(context.TODO(), scaleUpDeployment, metav1.UpdateOptions{})
 			Expect(err).NotTo(HaveOccurred())
 
 			waitForScaleUpToComplete(cs, ns, targetNodeCount)
```

Comment:
Because once applying for a lot of nodes like 20 one time, our portal has the limitation that for this situation, at most 40 nodes (with 2 CPUs per node) are allowed. So I set "expectedNode" to 38 in this situation, otherwise targetNodeCount will be 42 and it will fail in line 326.
(If applying for only several nodes like 5, the limitation is gone.)